hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ABucketExistence.java

```diff
@@ -0,0 +1,119 @@
+import static org.apache.hadoop.fs.contract.ContractTestUtils.writeDataset;
+import static org.apache.hadoop.fs.s3a.Constants.FS_S3A;
+import static org.apache.hadoop.fs.s3a.Constants.S3A_BUCKET_PROBE;
+import static org.apache.hadoop.test.LambdaTestUtils.intercept;
+
+/**
+ * Class to test bucket existence api.
+ * See {@link S3AFileSystem#doBucketProbing()}.
+ */
+public class ITestS3ABucketExistence extends AbstractS3ATestBase {
+
+  private FileSystem fs;
+
+  private final String randomBucket =
+          "random-bucket-" + UUID.randomUUID().toString();
+
+  private final URI uri = URI.create(FS_S3A + "://" + randomBucket);
+
+  @Test
+  public void testNoBucketProbing() throws Exception {
+    Configuration configuration = getConfiguration();
+    configuration.setInt(S3A_BUCKET_PROBE, 0);
+    try {
+      fs = FileSystem.get(uri, configuration);
+    } catch (IOException ex) {
+      LOG.error("Exception : ", ex);
+      throw ex;
+    }
+
+    Path path = new Path(uri);
+    intercept(FileNotFoundException.class,
+            "No such file or directory: " + path,
+        () -> fs.getFileStatus(path));
+
+    Path src = new Path(fs.getUri() + "/testfile");
+    byte[] data = dataset(1024, 'a', 'z');
+    intercept(FileNotFoundException.class,
+            "The specified bucket does not exist",
+        () -> writeDataset(fs, src, data, data.length, 1024 * 1024, true));
+  }
+
+  @Test
+  public void testBucketProbingV1() throws Exception {
+    Configuration configuration = getConfiguration();
+    configuration.setInt(S3A_BUCKET_PROBE, 1);
+    intercept(FileNotFoundException.class,
+        () -> FileSystem.get(uri, configuration));
+  }
+
+  @Test
+  public void testBucketProbingV2() throws Exception {
+    Configuration configuration = getConfiguration();
+    configuration.setInt(S3A_BUCKET_PROBE, 2);
+    intercept(FileNotFoundException.class,
+        () -> FileSystem.get(uri, configuration));
+  }
+
+  @Test
+  public void testBucketProbingParameterValidation() throws Exception {
+    Configuration configuration = getConfiguration();
+    configuration.setInt(S3A_BUCKET_PROBE, 3);
+    intercept(IllegalArgumentException.class,
+            "Value of " + S3A_BUCKET_PROBE + " should be between 0 to 2",
+            "Should throw IllegalArgumentException",
+        () -> FileSystem.get(uri, configuration));
+    configuration.setInt(S3A_BUCKET_PROBE, -1);
+    intercept(IllegalArgumentException.class,
+            "Value of " + S3A_BUCKET_PROBE + " should be between 0 to 2",
+            "Should throw IllegalArgumentException",
+        () -> FileSystem.get(uri, configuration));
+  }
+
+  @Override
+  protected Configuration getConfiguration() {
+    Configuration configuration = super.getConfiguration();
+    S3ATestUtils.disableFilesystemCaching(configuration);
+    return configuration;
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    IOUtils.cleanupWithLogger(getLogger(), fs);
```

Comment:
Adding this extra cleanup is throwing FileSystem is closed! because of this call AbstractFSContractTestBase.deleteTestDirInTeardown() in the super class teardown after each test

Reply:
ok. So what's happening then is that your tests are picking up a shared FS instance, not the one you've just configured with different bucket init settings. your tests aren't doing what you think they are

Reply:
My tests are running new FS instance only. I confirmed that using the IDE debugger. I think what is happening is, we are calling fs.close() twice one with the shared instance and other on my private instance which is stopping all the services for a particular Fs leading to mismatch.

Reply:
Found the issue. Rather than overriding the teardown method I implemented it which caused the Junit to call teardown() twice thus cause all the above problems. 
Sorry My Bad. :(

Reply:
no worries